src/fty-security-wallet.service.in

```diff
@@ -13,6 +13,14 @@ EnvironmentFile=-@sysconfdir@/default/fty
 EnvironmentFile=-@sysconfdir@/default/fty__%n.conf
 Environment="prefix=@prefix@"
 ExecStart=@prefix@/bin/fty-security-wallet -c @sysconfdir@/@PACKAGE@/fty-security-wallet.cfg
+
+# This service is only active, as far as consumers are concerned,
+# after it listens on the Unix socket used to interact with it.
+ExecStartPost=/bin/dash -c "while sleep 0.1; do test -S \"@socketSecurityWallet@\" || continue; PIDS=\"`/bin/fuser \"@socketSecurityWallet@\" 2>/dev/null | sed -e 's,^ *,,' -e 's, *$,,'`\" && [ -n \"$PIDS\" ] || continue ; ( for P in $PIDS ; do ls -la /proc/$P/exe ; done | egrep ' @prefix@/bin/fty-security-wallet$' ) && break ; rePIDS=\"(`echo \"$PIDS\" | sed 's,  *,|,g'`)\" ; ps -ef | grep -v grep | grep -E \"$rePIDS\" && break ; done; echo 'Socket is being listened'"
```

Comment:
Isn't that obsoleted by the `sd_notify()` call?

Reply:
"Yes and no". For one, so far I do not trust systemd enough to do a proper job, so a fallback should not hurt, and only adds a small load on top - e.g. not waiting several seconds after ExecStart is deemed done because it forked (in original codebase + PR #49), but waiting in normal case his one loop cycle to see that socket already exists, because ExecStart is only done after sd_notify... (oh, wait, Type is still not fixed!)

For another, this PR is explicitly stated as a merger of two prior attacks on this problem. As such, it can't only include one of those (incomplete or inefficient alone) solutions, right?

Having this checker around also allows to copy-paste its waiting code to another service manage framework if we decide to drift away from systemd :D

It has failed too often on us, blocking in systemctl never reaching its message bus, etc.

Reply:
UPDATE: Type=notify should now be fixed, and depending on use or not of libsystemd-dev(el) (optional in this PR)